clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -785,15 +826,23 @@ raw2trace_t::create_instr_summary(void *tls, uint64 modidx, uint64 modoffs,
     if (block == nullptr) {
         block = new block_summary_t(block_start, instr_count);
         DEBUG_ASSERT(index >= 0 && index < static_cast<int>(block->instrs.size()));
-        hashtable_add(&decode_cache_[tdata->worker], block_start, block);
-        VPRINT(5, "Created new block summary " PFX " for " PFX "\n", block, block_start);
+        decode_cache_[tdata->worker].add(modidx, modoffs, block);
+        VPRINT(5,
+               "Created new block summary " PFX " for " PFX " modidx=" INT64_FORMAT_STRING
+               " modoffs=" HEX64_FORMAT_STRING "\n",
+               block, block_start, modidx, modoffs);
         tdata->last_decode_block_start = block_start;
```

Comment:
I think this is asking the same question as the other comment: this is because there is no sentinel.  There is a comment in the header.  I will duplicate the comment here too.